### ISSUE TITLE
Unit test fails with EXC_BAD_INSTRUCTION on swift_2.3 branch

### ISSUE THREAD

**Issue by nuudles** [hkellaway/Gloss#208]
I encountered a crash when encoding JSON on my updated project (to Swift 2.3) that uses Gloss built with Carthage. I was finally able to reproduce it in the unit tests, but only in `Release` mode, which Carthage uses when building the frameworks. I was able to further narrow it down to the `Swift Compiler - Code Generation` `Optimization Level` setting. When this is set to `None` for the Gloss framework, all unit tests pass, however, when it is set to `Fast, Whole Module Optimization`, I get an `EXC_BAD_INSTRUCTION` as seen in this screenshot:

![image](https://cloud.githubusercontent.com/assets/741746/18594001/e9bb7476-7c0b-11e6-87d4-acdf1c9fae16.png)

The test that causes the crash is `testEncodeEncodableDictionary` in `EncoderTests.swift`. I can workaround the issue by setting the `Optimization Level` to none for `Release` for now, but would like a long term solution for the problem.

**Comment by hkellaway** [comment]
this is the same issue listed here https://github.com/hkellaway/Gloss/issues/135 - which i've yet to find a solution for. that `setValue` implementation is intended to allow nested keys paths (e.g. `self.userID = "account.user.id" <~~ json`), but can exhibit a crash on Release. 

i'm considering removing simply removing support for nested keypaths in the next major version due to it.

if you need to re-work any nested keypaths you can - 1. create nested models 2. create custom transformations, e.g.

``` swift
extension Decoder {

    static func decodeUserID(key: String, json: JSON) -> String? {

        guard let userID = json["account"]["user"]["id"] as? String else {
            return nil
        }

        return userID
    }

}
```

**Comment by hkellaway** [comment]
@nuudles you mentioned settings the optimization level to None - did you mean for the framework or for your application? if the latter - i instead tried setting the optimization level to None for the framework and this seems to address the crash. do you see an issue with that solution?

that update is available on branch `swift_2.3` if you wanted to try it out

**Comment by nuudles** [comment]
@hkellaway I set the optimization level to `None` for the framework, which enables me to use the framework for now, but I don't believe it to be a long-term solution; I'd rather have the optimization settings set for Release mode as recommended by Apple.

**Comment by hkellaway** [comment]
`1.0.0` was just released, as compatible with Swift 3.0.  however, i was not comfortable including a feature that caused crashes at runtime. therefore, nested keypaths have been removed.

apologies if you have a lot of nested keypaths!  See more here https://github.com/hkellaway/Gloss#nested-keypaths-deprecation

**Comment by hkellaway** [comment]
nested keypaths was reintroduced in the latest version `1.1.0` 🎉
